new-dti-website/src/app/page.tsx

```diff
@@ -1,11 +1,108 @@
-import { Button } from '../../components/ui/button';
-import Navbar from '../../components/navbar';
-
-const Home = (): JSX.Element => (
-  <div>
-    <Navbar />
-    <Button>This is a ShadCN Button.</Button>
-  </div>
-);
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import Icon from '../../components/icons';
+import Slideshow from '../../components/slideshow';
+
+const Home: React.FC = () => {
+  const [selectedIcon, setSelectedIcon] = useState<number | null>(null);
+  const [timer, setTimer] = useState<NodeJS.Timeout | null>(null);
+
+  const icons = [
+    {
+      src: '/images/DTI_notsel.png',
+      hover: '/images/DTI_hover.png',
+      active: '/images/DTI_current.png',
+      altText: 'DTI',
+      width: 80,
+      height: 80
+    },
+    {
+      src: '/images/Family_notsel.png',
+      hover: '/images/Family_hover.png',
+      active: '/images/Family_current.png',
+      altText: 'Family',
+      width: 80,
+      height: 80
+    },
+    {
+      src: '/images/Collaboration_notsel.png',
+      hover: '/images/Collaboration_hover.png',
+      active: '/images/Collaboration_current.png',
+      altText: 'Collaboration',
+      width: 100,
+      height: 80
+    },
+    {
+      src: '/images/Events_notsel.png',
+      hover: '/images/Events_hover.png',
+      active: '/images/Events_current.png',
+      altText: 'Events',
+      width: 90,
+      height: 90
+    },
+    {
+      src: '/images/Initiatives_notsel.png',
+      hover: '/images/Initiatives_hover.png',
+      active: '/images/Initiatives_current.png',
+      altText: 'Initiatives',
+      width: 80,
+      height: 80
+    }
+  ];
+
+  useEffect(() => {
+    if (timer) clearTimeout(timer);
+
+    setTimer(
+      setTimeout(() => {
+        if (selectedIcon === null || selectedIcon >= icons.length - 1) {
+          setSelectedIcon(0);
+        } else {
+          setSelectedIcon(selectedIcon + 1);
+        }
+      }, 3000)
+    );
+
+    return () => {
+      if (timer) clearTimeout(timer);
+    };
+  }, [selectedIcon]);
```

Comment:
Try adding the dependencies to the dependency array as mentioned in the warning. If they caused undesired behavior, you can remove and then suppress the warning: https://codedamn.com/news/reactjs/useeffect-dependency